src/lib/core/components/computations/plugins/abundance.tsx

```diff
@@ -24,11 +24,52 @@ export const AbundanceConfig = t.type({
 export const plugin: ComputationPlugin = {
   configurationComponent: AbundanceConfiguration,
   configurationDescriptionComponent: AbundanceConfigDescriptionComponent,
-  visualizationTypes: {
-    boxplot: boxplotVisualization,
-    scatterplot: scatterplotVisualization,
-  },
   createDefaultComputationSpec: createDefaultComputationSpec,
+  visualizationPlugins: {
+    boxplot: boxplotVisualization.withOptions({
+      getXAxisVariable(config) {
+        if (AbundanceConfig.is(config)) {
+          return config.collectionVariable;
+        }
+      },
+      getComputedYAxisDetails(config) {
+        if (AbundanceConfig.is(config)) {
+          return {
+            entityId: config.collectionVariable.entityId,
+            placeholderDisplayName: 'Relative abundance',
+          };
+        }
+      },
+      getPlotSubtitle(config) {
+        if (AbundanceConfig.is(config)) {
+          return `Ranked abundance: Variables with ${config.rankingMethod} = 0 removed. Showing up to the top ten variables.`;
+        }
+      },
+      hideShowMissingnessToggle: true,
+    }),
+    scatterplot: scatterplotVisualization.withOptions({
+      getComputedYAxisDetails(config) {
+        if (AbundanceConfig.is(config)) {
+          return {
+            entityId: config.collectionVariable.entityId,
+            placeholderDisplayName: 'Relative abundance',
+          };
+        }
+      },
+      getOverlayVariable(config) {
```

Comment:
That was the (half-baked) idea. We could just as easily do something link this:

```typescript
interface ComputedVariableDetails {
  type: 'computed';
  entityId: string;
  placeholderDisplayName: string;
}

interface CollectionVariableDetails {
  type: 'collection';
  entityId: string;
  variableId: string;
}

interface XAxisVariableOptions {
  getXAxisVariable(config: unknown): ComputedVariableDetails | CollectionVariableDetails
}
```

This may be better and provide more flexibility in the future.

Reply:
oh yikes sorry about the weird italics above - glad you could parse what i meant!

If it's easy, i agree your suggestion adds flexibility and i'd say it also adds a lot of clarity, so i'm a big fan!

Reply:
I think we should just stick with what we have for now. If we find that we need to support either computed or collection vars for a particular axis, we can revisit. Does that work for you @asizemore?

Reply:
works for me!